### PR TITLE
[applications] Add CalculateLumpedMassVector method

### PR DESCRIPTION
**Pull request by riccardotosi**
**Description**
This PR follows #7944 and #7934 discussion.

I understand this PR is the base PR to which we should add changes for elements using the explicit builder. Right now only the explicit convection diffusion and explicit compressible come to my mind. Am I right? For this reason, I will make this a draft PR. Please let me know if I misunderstood something.

@rubenzorrilla, do you take care of the compressible element, or do you want me to do it?

**Please mark the PR with appropriate tags:**

- Enhancement, Applications, C++, Incomplete

**Changelog**
Please summarize the changes in one list to generate the changelog:

- Method CalculateLumpedMassVector is added wherever needed.

### CODE REVIEW

**Comment by rubenzorrilla** [comment]
I can do the compressible Navier-Stokes element changes. Considering that this feature has been recently merged to the core I don't think there is any other place where this is being used. In any case, the base element error will pop up so I think we can safely proceed with the strategy changes once this is merged.

File: applications/ConvectionDiffusionApplication/custom_elements/qs_convection_diffusion_explicit.cpp [hunk 1]
```diff
@@ -231,6 +231,8 @@ void QSConvectionDiffusionExplicit<2,3>::CalculateLumpedMassVector(
     for (IndexType i = 0; i < local_size; ++i) {
         rLumpedMassVector(i) = one_third;
     }
+    // Assumption all the Gauss points have the same weight, so we multiply by the volume
+    rLumpedMassVector *= GetGeometry().Volume();
```
Comment (rubenzorrilla) [inline applications/ConvectionDiffusionApplication/custom_elements/qs_convection_diffusion_explicit.cpp]: Why not multiplying the `one_third` by the volume. Internally, the  `*=` is multiplying each component of the vector. Considering that this is an explicit strategy any optimization becomes relevant.

Comment (riccardotosi) [reply applications/ConvectionDiffusionApplication/custom_elements/qs_convection_diffusion_explicit.cpp]: You are right

**Comment by riccardotosi** [comment]
Ping

File: applications/ConvectionDiffusionApplication/custom_elements/qs_convection_diffusion_explicit.cpp [hunk 2]
```diff
@@ -214,6 +214,53 @@ void QSConvectionDiffusionExplicit<3,4>::CalculateMassMatrix(
...
+    // Initialize and calculate elemental lumped mass vector
+    if (rLumpedMassVector.size() != local_size) {
+        rLumpedMassVector.resize(local_size, false);
+    }
+    for (IndexType i = 0; i < local_size; ++i) {
+        rLumpedMassVector(i) = one_fourth_volume;
+    }
+
+    KRATOS_CATCH("");
+}
```
Comment (rubenzorrilla) [inline applications/ConvectionDiffusionApplication/custom_elements/qs_convection_diffusion_explicit.cpp]: You don't need the template specialization in this case. Check the implementation in the compressible element.

Comment (riccardotosi) [reply applications/ConvectionDiffusionApplication/custom_elements/qs_convection_diffusion_explicit.cpp]: Thanks. Your implementation is way more clever than mine.. I will push it now.

File: applications/FluidDynamicsApplication/symbolic_generation/compressible_navier_stokes/compressible_navier_stokes_explicit_cpp_template_with_integration.cpp [hunk 3]
```diff
@@ -1535,6 +1535,22 @@ void CompressibleNavierStokesExplicit<3,4>::CalculateMassMatrix(
...
+    const ProcessInfo& rCurrentProcessInfo) const
+{
+    // Initialize the lumped mass vector
+    constexpr IndexType size = TNumNodes * BlockSize;
+    if (rLumpedMassVector.size() != BlockSize) {
+        rLumpedMassVector.resize(size, false);
+    }
+
+    // Fill the lumped mass vector
+    const double nodal_mass = GetGeometry().DomainSize() / TNumNodes;
```
Comment (riccardotosi) [inline applications/FluidDynamicsApplication/symbolic_generation/compressible_navier_stokes/compressible_navier_stokes_explicit_cpp_template_with_integration.cpp:1550]: Why not multiplying by volume/area?

Comment (rubenzorrilla) [reply applications/FluidDynamicsApplication/symbolic_generation/compressible_navier_stokes/compressible_navier_stokes_explicit_cpp_template_with_integration.cpp:1550]: `DomainSize` gives you back either the volume or area according to the geometry working space dimension. That is, area for 2D geometries and volume for 3D volumetric geometries.

Comment (riccardotosi) [reply applications/FluidDynamicsApplication/symbolic_generation/compressible_navier_stokes/compressible_navier_stokes_explicit_cpp_template_with_integration.cpp:1550]: Thanks for specifying.

**Comment by rubenzorrilla** [comment]
@KlausBSautter I updated your `CableNet` elements too. Please check that I didn't mess up something.

**Comment by rubenzorrilla** [comment]
BTW, this should fix the hiding overload virtual functions warnings.

**Comment by KlausBSautter** [comment]
@rubenzorrilla I am on Holidays. I can Check it in 2 Weeks.

**Comment by rubenzorrilla** [comment]
> @rubenzorrilla I am on Holidays. I can Check it in 2 Weeks.
> But I do not want you to wait for me. As Long as the Tests run it should be ok

I've just compiled your application and ran the tests without any problem. If you trust us we would proceed with the merging :smile: (@riccardotosi needs this to continue his developments).

**Comment by riccardotosi** [comment]
Can we merge @rubenzorrilla @KlausBSautter?

**Comment by rubenzorrilla** [comment]
I think so, but we need someone to approve it.

**Comment by mrhashemi** [comment]
As far as I see, _GetGeometry().DomainSize() / TNumNodes_ is calculated as the entries of the lumped-mass matrix (stored in a vector). If there is no conflict, it is OK in my opinion.

**Review by mrhashemi — APPROVED** [review]

**Comment by KlausBSautter** [comment]
If all the tests are running it is ok. I cannot check it myself right now.

**Comment by riccardotosi** [comment]
Hi @KlausBSautter , I understand Ruben tried your tests within this branch, and they run ok.

**Pull request merged**